### PR TITLE
refactor drools processing in ParticipantToLegacyClient class

### PR DESCRIPTION
**Pull request by jimlebeau**
Refactor ParticipantToLegacyClient class so that messages from DroolsService are passed to the MessageBuilder in order to properly process errors from Drools.

## Description

Drools implementation for R-10971 and R-10975 on the CSEC History entity.

- ParticipantToLegacyClient.java was modified so that CSEC History DocTool rules are enforced using the DroolsService class.

- - saveOrUpdateCsec() method is modified to use the DroolsService.performBusinessRules() method after the CSEC History object is created.
- - saveOrUpdateCsec() method is modified so that any messages returned by DroolsService are passed to the MessageBuilder.
- - isValidCsecs() method is modified so that R10971CsecEndDate.java class is not used.
- created R-10971.drl to encapsulate the DocTool rule for CSEC History.
- modified R-10975.drl to use the CSEC History entity class instead of the CSEC domain class.

- created CsecHistoryBusinessRulesValidationTest.java that will test the implementation of the R-10971 and R-1095 DocTool rules (commercial-sexual-exploitation-agenda".
- removed R10975Test.java as it is redundent to the CsecBusinessRulesValidationTest class
- remove R10971CsecEndDate.java because DocTool rule is now implemented in R-10971.drl
- remove R10971CsecEndDateTest.java

## Jira Issue link

N/A

## Tests
- [X] I have included unit tests 
- [X] I have included functional tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the google code style of this project.
- [X] I have ran all tests for the project.
- [X] My code is in a stable state ready to be deployable, but not necessarily complete.
- [X] I promise on my honor as a CWDS developer to ensure I don't break the build, to check SonarQube after the build completes, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/rest/services/ParticipantToLegacyClient.java [hunk 1]
```diff
@@ -139,6 +137,7 @@ public ParticipantToLegacyClient(ClientService clientService,
   public ClientParticipants saveParticipants(ScreeningToReferral screeningToReferral,
       String dateStarted, String timeStarted, String referralId, MessageBuilder messageBuilder) {
     ClientParticipants clientParticipants = new ClientParticipants();
+    this.droolsService = new DroolsService();
```
Comment (AVandana-OSI) [inline src/main/java/gov/ca/cwds/rest/services/ParticipantToLegacyClient.java]: Please remove this from the constructor as we have Injected through Guice, we don't need to specify it in the Constructor again,

File: src/test/java/gov/ca/cwds/rest/validation/rules/CsecHistoryBusinessRulesValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package gov.ca.cwds.rest.validation.rules;
+
+
```
Comment (AVandana-OSI) [inline src/test/java/gov/ca/cwds/rest/validation/rules/CsecHistoryBusinessRulesValidationTest.java]: Please separate the JUnit for each rule with a number, we all decided to keep that way a long way. Please change it.

File: src/main/resources/rules/commercial-sexual-exploitation-history/R-10971.drl [hunk 3]
```diff
@@ -0,0 +1,31 @@
+//created on: Oct 15, 2018
+import gov.ca.cwds.data.legacy.cms.entity.CsecHistory;
+import gov.ca.cwds.data.legacy.cms.entity.syscodes.SexualExploitationType;
+import gov.ca.cwds.rest.exception.IssueDetails
+
+global java.util.Set validationDetailsList
+
+/*
+   DocTool Rule: R - 10971
+
```
Comment (AVandana-OSI) [inline src/main/resources/rules/commercial-sexual-exploitation-history/R-10971.drl]: Let's keep the whole Access Logic and rule text it exists.

File: src/main/java/gov/ca/cwds/rest/services/ParticipantToLegacyClient.java (hunk 1 above)
Comment (jimlebeau) [reply src/main/java/gov/ca/cwds/rest/services/ParticipantToLegacyClient.java]: modified to construct DroolsService in test class and use setDroolsService() method of ParticipantToLegacyClient class

File: src/main/resources/rules/commercial-sexual-exploitation-history/R-10971.drl (hunk 3 above)
Comment (jimlebeau) [reply src/main/resources/rules/commercial-sexual-exploitation-history/R-10971.drl]: modified to include access logic

File: src/test/java/gov/ca/cwds/rest/validation/rules/CsecHistoryBusinessRulesValidationTest.java (hunk 2 above)
Comment (jimlebeau) [reply src/test/java/gov/ca/cwds/rest/validation/rules/CsecHistoryBusinessRulesValidationTest.java]: Class is modeled on CrossReportBusinessRulesValidationTest class and SafelySurrenderBabiesBusinessRulesValidationTest class in same package.  Class will test all drools of "commercial-sexual-exploitation-history" agenda-group.  Will keep it this way until these other classes are refactored.

**Comment by AVandana-OSI** [comment]
retest this please

**Pull request merged**